backend/clubs/serializers.py

```diff
@@ -994,30 +994,48 @@ def get_fields(self):
 
     def to_representation(self, instance):
         """
-        Return the previous approved version of a club for users
-        that should not see unapproved content.
+        Return appropriate version of club based on user permissions and club status:
+        - Privileged users see current version (or when bypass=True)
```

Comment:
What's the use case for `bypass=True`? I don't think clubs should be allowed to show an unapproved version at all

Reply:
I think we use `bypass=True` when someone's accepting a membership invite to a club that hasn't been approved yet. (Basing this off the comments in `ClubPermission` from `permissions.py`.)